---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# hgnc <img src='man/figures/logo.svg' align="right" height="139" />

<!-- badges: start -->
[![CRAN status](https://www.r-pkg.org/badges/version/hgnc)](https://CRAN.R-project.org/package=hgnc)
<!-- badges: end -->

The goal of `{hgnc}` is to easily import Human Gene Nomenclature into R.

## Installation

Install `{hgnc}` from CRAN:

``` r
install.packages("hgnc")
```

## Usage

### Basic usage

To import the latest HGNC gene data set in tabular format directly into memory
as a tibble do as follows:

```{r}
library(hgnc)
# Check the date of HGNC last update
last_update()
# Import the data set in tidy tabular format
(hgnc_dataset <- import_hgnc_dataset())
```

### Convert between gene identifiers

Use `crosswalk()` to convert values found in one of the columns of an HGNC gene
data set to values in another:

```{r}
# convert a set of hgnc_ids to symbols
hgnc_ids <- c("HGNC:44948", "HGNC:43240", "HGNC:23357", "HGNC:1855", "HGNC:39400")
crosswalk(hgnc_ids, from = 'hgnc_id', to = 'symbol')

# convert a set of entrez_ids to ensembl_gene_ids
entrez_ids <- c(79933, 109623458, 158471, 54987, 81631)
crosswalk(entrez_ids, from = 'entrez_id', to = 'ensembl_gene_id')
```

## Motivation

You could go to [www.genenames.org](https://www.genenames.org) and download the
files yourself. So why the need for this R package?

`{hgnc}` really is just a convenience package. The main advantage is that the
function `import_hgnc_dataset()` reads in the data in tabular format with
all the columns with the appropriate type (so you don't have to specify it
yourself). As an extra step, those variables that contain multiple values are
encoded as list-columns.

Remember that list-columns can be expanded with `tidyr::unnest()`. E.g.,
`alias_symbol` is a list-column containing multiple alternative aliases to the
standard `symbol`:

```{r}
hgnc_dataset |>
  dplyr::filter(symbol == 'TP53') |>
  dplyr::select(c('symbol', 'alias_symbol'))

hgnc_dataset |>
  dplyr::filter(symbol == 'TP53') |>
  dplyr::select(c('symbol', 'alias_symbol')) |>
  tidyr::unnest(cols = 'alias_symbol')
```

In addition, we also provide the function `filter_by_keyword()` that allows
filtering the data set based on a keyword or regular expression. By default this
function will look into all columns that contain gene symbols or names
(`symbol`, `name`, `alias_symbol`, `alias_name`, `prev_symbol` and `prev_name`).
It works automatically with list-columns too.

Look for entries in the data set that contain the keyword `"TP53"`:

```{r}
hgnc_dataset |>
  filter_by_keyword('TP53') |>
  dplyr::select(1:4)
```

Restrict the search to the `symbol` column:

```{r}
hgnc_dataset |>
  filter_by_keyword('TP53', cols = 'symbol') |>
  dplyr::select(1:4)
```

Search for the whole word `"TP53"` exactly by taking advantage of regular
expressions:

```{r}
hgnc_dataset |>
  filter_by_keyword('^TP53$', cols = 'symbol') |>
  dplyr::select(1:4)
```

## Citing the HGNC

To cite HGNC nomenclature resources use:

- Tweedie S, Braschi B, Gray KA, Jones TEM, Seal RL, Yates B, Bruford EA. _Genenames.org: the HGNC and VGNC resources in 2021._ Nucleic Acids Res. 49, D939--D946 (2021). doi: [10.1093/nar/gkaa980](https://doi.org/10.1093/nar/gkaa980)

To cite data within the database use the following format:
 
- HGNC Database, HUGO Gene Nomenclature Committee (HGNC), European Molecular Biology Laboratory, European Bioinformatics Institute (EMBL-EBI), Wellcome Genome Campus, Hinxton, Cambridge CB10 1SD, United Kingdom www.genenames.org. 

Please include the month and year you retrieved the data cited.
