<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Import HGNC data — import_hgnc_dataset • hgnc</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet"><link href="../deps/Inconsolata-0.4.10/font.css" rel="stylesheet"><link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Import HGNC data — import_hgnc_dataset"><meta name="description" content="import_hgnc_dataset() imports HGNC data into R. Specify a directory path
in addition if you wish the save the data to disk."><meta property="og:description" content="import_hgnc_dataset() imports HGNC data into R. Specify a directory path
in addition if you wish the save the data to disk."><meta property="og:image" content="https://www.pattern.institute/hgnc/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hgnc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.9003</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/import.html">Import Human Gene Nomenclature</a></li>
    <li><a class="dropdown-item" href="../articles/mapping.html">Crosswalk across identifiers</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/patterninstitute/hgnc/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Import HGNC data</h1>
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/hgnc/blob/main/R/import_hgnc_dataset.R" class="external-link"><code>R/import_hgnc_dataset.R</code></a></small>
      <div class="d-none name"><code>import_hgnc_dataset.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>import_hgnc_dataset()</code> imports HGNC data into R. Specify a directory <code>path</code>
in addition if you wish the save the data to disk.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">import_hgnc_dataset</span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="latest_archive_url.html">latest_archive_url</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-file">file<a class="anchor" aria-label="anchor" href="#arg-file"></a></dt>
<dd><p>A file or URL of the complete HGNC data set (in TSV format).
Use <code><a href="list_archives.html">list_archives()</a></code> to list previous versions of these data. Pass one
of the URLs (column <code>url</code>) to <code>file</code> to import that specific version. By
default the value of <code>file</code> is the URL corresponding to the latest version,
i.e. the returned value of <code><a href="latest_archive_url.html">latest_archive_url()</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <a href="https://tibble.tidyverse.org/reference/tibble-package.html" class="external-link">tibble</a> of the HGNC data set consisting of
55 columns:</p><ul><li><p><code>hgnc_id</code>: A unique ID provided by HGNC for each gene with an approved symbol. IDs are of the format <code>'HGNC:n'</code>, where <code>n</code> is a unique number. HGNC IDs remain stable even if a name or symbol changes.</p></li>
<li><p><code>hgnc_id2</code>: A stripped down version of <code>hgnc_id</code> where the prefix <code>'HGNC:'</code> has been removed. This column is added by the package <code>{hgnc}</code>.</p></li>
<li><p><code>symbol</code>: The official gene symbol approved by the HGNC, typically a short form of the gene name. Symbols are approved in accordance with the Guidelines for Human Gene Nomenclature.</p></li>
<li><p><code>name</code>: The full gene name approved by the HGNC; corresponds to the approved symbol above.</p></li>
<li><p><code>locus_group</code>: A group name for a set of related locus types as defined by the HGNC. One of: <code>'protein-coding gene'</code>, <code>'non-coding RNA'</code>, <code>'pseudogene'</code> or <code>'other'</code>.</p></li>
<li><p><code>locus_type</code>: Specifies the genetic class of each gene entry, including various types of RNA and other gene-related categories, such as pseudogenes and virus integration sites.</p></li>
<li><p><code>status</code>: Status of the symbol report, which can be either <code>'Approved'</code> or <code>'Entry Withdrawn'</code>.</p></li>
<li><p><code>location</code>: Chromosomal location. Indicates the cytogenetic location of the gene or region on the chromosome, e.g., <code>'19q13.43'</code>. In the absence of that information, it may be listed as <code>'not on reference assembly'</code>, <code>'unplaced'</code>, or <code>'reserved'</code>.</p></li>
<li><p><code>location_sortable</code>: A sortable version of the <code>location</code> column, allowing easier sorting by chromosomal location.</p></li>
<li><p><code>alias_symbol</code>: Alternative symbols that have been used to refer to the gene. Aliases may be from literature, other databases, or represent membership of a gene group.</p></li>
<li><p><code>alias_name</code>: Alternative names for the gene. Aliases may be from literature, other databases, or represent membership of a gene group.</p></li>
<li><p><code>prev_symbol</code>: This field displays any symbols that were previously HGNC-approved nomenclature.</p></li>
<li><p><code>prev_name</code>: This field displays any names that were previously HGNC-approved nomenclature.</p></li>
<li><p><code>gene_group</code>: A gene group. Each gene has been assigned to one or more groups, according to either sequence similarity or information from publications, specialist advisors, or other databases.</p></li>
<li><p><code>gene_group_id</code>: Gene group identifier, an integer number. This column contains the gene group identifiers. See <code>gene_group</code> for the gene group name.</p></li>
<li><p><code>date_approved_reserved</code>: The date the entry was first approved.</p></li>
<li><p><code>date_symbol_changed</code>: The date the gene symbol was last changed.</p></li>
<li><p><code>date_name_changed</code>: The date the gene name was last changed.</p></li>
<li><p><code>date_modified</code>: Date the entry was last modified.</p></li>
<li><p><code>entrez_id</code>: Entrez gene identifier.</p></li>
<li><p><code>ensembl_gene_id</code>: Ensembl gene identifier.</p></li>
<li><p><code>vega_id</code>: VEGA gene identifier.</p></li>
<li><p><code>ucsc_id</code>: UCSC gene identifier.</p></li>
<li><p><code>ena</code>: International Nucleotide Sequence Database Collaboration (GenBank, ENA and DDBJ) accession number(s).</p></li>
<li><p><code>refseq_accession</code>: The Reference Sequence (RefSeq) identifier for that entry, provided by the NCBI.</p></li>
<li><p><code>ccds_id</code>: Consensus CDS identifier.</p></li>
<li><p><code>uniprot_ids</code>: UniProt protein accession.</p></li>
<li><p><code>pubmed_id</code>: Pubmed and Europe Pubmed Central PMIDs.</p></li>
<li><p><code>mgd_id</code>: Mouse genome informatics database identifier.</p></li>
<li><p><code>rgd_id</code>: Rat genome database gene identifier.</p></li>
<li><p><code>lsdb</code>: The name of the Locus Specific Mutation Database and URL for the gene.</p></li>
<li><p><code>cosmic</code>: Symbol used within the Catalogue of somatic mutations in cancer for the gene.</p></li>
<li><p><code>omim_id</code>: Online Mendelian Inheritance in Man (OMIM) identifier.</p></li>
<li><p><code>mirbase</code>: miRBase identifier.</p></li>
<li><p><code>homeodb</code>: Homeobox Database identifier.</p></li>
<li><p><code>snornabase</code>: snoRNABase identifier.</p></li>
<li><p><code>bioparadigms_slc</code>: Symbol used to link to the SLC tables database at bioparadigms.org for the gene.</p></li>
<li><p><code>orphanet</code>: Orphanet identifier.</p></li>
<li><p><code>pseudogene_org</code>: Pseudogene.org identifier.</p></li>
<li><p><code>horde_id</code>: Symbol used within HORDE for the gene.</p></li>
<li><p><code>merops</code>: Identifier used to link to the MEROPS peptidase database.</p></li>
<li><p><code>imgt</code>: Symbol used within international ImMunoGeneTics information system.</p></li>
<li><p><code>iuphar</code>: The objectId used to link to the IUPHAR/BPS Guide to PHARMACOLOGY database.</p></li>
<li><p><code>kznf_gene_catalog</code>: Lawrence Livermore National Laboratory Human KZNF Gene Catalog (LLNL) identifier.</p></li>
<li><p><code>mamit_trnadb</code>: Identifier to link to the Mamit-tRNA database.</p></li>
<li><p><code>cd</code>: Symbol used within the Human Cell Differentiation Molecule database for the gene.</p></li>
<li><p><code>lncrnadb</code>: lncRNA Database identifier.</p></li>
<li><p><code>enzyme_id</code>: ENZYME EC accession number.</p></li>
<li><p><code>intermediate_filament_db</code>: Identifier used to link to the Human Intermediate Filament Database.</p></li>
<li><p><code>rna_central_ids</code>: Identifier in the RNAcentral, The non-coding RNA sequence database.</p></li>
<li><p><code>lncipedia</code>: The LNCipedia identifier to which the gene belongs. This will only appear if the gene is a long non-coding RNA.</p></li>
<li><p><code>gtrnadb</code>: The GtRNAdb identifier to which the gene belongs. This will only appear if the gene is a tRNA.</p></li>
<li><p><code>agr</code>: The Alliance of Genomic Resources HGNC ID for the Human gene page within the resource.</p></li>
<li><p><code>mane_select</code>: MANE Select nucleotide accession with version (i.e., NCBI RefSeq or Ensembl transcript ID and version).</p></li>
<li><p><code>gencc</code>: Gene Curation Coalition (GenCC) Database identifier.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="fu">import_hgnc_dataset</span><span class="op">(</span><span class="op">)</span> <span class="co"># \dontrun{}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.pattern.institute/team/rmagno" class="external-link">Ramiro Magno</a>, <a href="https://www.pattern.institute/team/iduarte" class="external-link">Isabel Duarte</a>, <a href="https://www.pattern.institute" class="external-link">Pattern Institute</a>.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer></div>





  </body></html>

